packages/flutter/lib/src/widgets/image.dart

```diff
@@ -459,6 +551,182 @@ class Image extends StatefulWidget {
+  ///
+  /// To have finer-grained control over the way that an image's loading
+  /// progress is communicated to the user, see [loadingBuilder].
+  ///
+  /// ## Chaining with [loadingBuilder]
+  ///
+  /// If a [loadingBuilder] has _also_ been specified for an image, the two
+  /// builders will be chained together, and the _result_ of this builder will
+  /// be passed as the `child` argument to the [loadingBuilder]. For example,
+  /// consider the following builders used in conjunction:
+  ///
+  /// ```dart
+  /// Image(
+  ///   ...
+  ///   frameBuilder: (BuildContext context, Widget child, int frame, bool wasSynchronouslyLoaded) {
+  ///     return Padding(
+  ///       padding: EdgeInsets.all(8.0),
+  ///       child: child,
+  ///     );
+  ///   },
+  ///   loadingBuilder: (BuildContext context, Widget child, ImageChunkEvent loadingProgress) {
+  ///     return Center(child: child);
+  ///   },
+  /// )
+  /// ```
+  ///
+  /// In this example, the widget hierarchy will contain the following:
+  ///
+  /// ```dart
+  /// Center(
+  ///   Padding(
+  ///     padding: EdgeInsets.all(8.0),
+  ///     child: <image>,
+  ///   ),
+  /// )
+  /// ```
+  ///
+  /// {@tool snippet --template=stateful_widget_animation}
+  ///
+  /// The following sample demonstrates how to use this builder to implement an
+  /// image that fades in once it's been loaded.
+  ///
+  /// This sample contains a limited subset of the functionality that the
+  /// [FadeInImage] widget provides out of the box.
+  ///
+  /// ```dart
+  /// @override
+  /// Widget build(BuildContext context) {
+  ///   return Image.network(
+  ///     'https://example.com/image.jpg',
+  ///     frameBuilder: (BuildContext context, Widget child, int frame, bool wasSynchronouslyLoaded) {
+  ///       Widget result = child;
+  ///       if (_controller.isAnimating || (frame == 0 && !wasSynchronouslyLoaded)) {
+  ///         result = FadeTransition(
+  ///           opacity: _controller,
+  ///           child: result,
+  ///         );
+  ///         if (!_controller.isAnimating) {
+  ///           _controller.animateTo(1.0, curve: Curves.easeOut);
+  ///         }
+  ///       }
+  ///       return result;
+  ///     },
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  ///
+  /// Run against a real-world image, the previous example renders the following
+  /// image.
+  ///
+  /// {@animation 400 400 https://flutter.github.io/assets-for-api-docs/assets/widgets/frame_builder_image.mp4}
+  final ImageFrameBuilder frameBuilder;
+
+  /// A builder that specifies the widget to display to the user while an image
+  /// is still loading.
+  ///
+  /// If this is null, and the image is loaded incrementally (e.g. over a
+  /// network), the user will receive no indication of the progress as the
+  /// bytes of the image are loaded.
+  ///
+  /// Once an image has fully loaded, this builder will no longer be consulted.
```

Comment:
fully loaded for a multi-frame image means that all bytes for all frames are loaded, correct? Maybe call this out as people may be confused why this is called for a multi-frame image after the first frame is shown?

Reply:
This actually should have been removed, since the builder is always consulted now.

Done.